src/transformation.js

```diff
@@ -778,6 +784,19 @@ class Transformation extends TransformationBase {
     return this.param(value, "page", "pg");
   }
 
+  placeholder(value) {
+    const mode = PLACEHOLDER_IMAGE_MODES[value] || PLACEHOLDER_IMAGE_MODES.blur;
+    const width = this.getValue("width");
+    const height = this.getValue("height");
+    let transformations = mode({width, height});
```

Comment:
Only predominant accepts width and height and according to the spec and the ref implementation, it doesn't have to. 
If you do, we do the manipulation on a single pixel. If not, we do another calc. 

In addition, the entire code that go to consts file and the returned value is a method is strange. 
it took me 7 minutes to understand what's going on here.

Reply:
Converted from functions to just arrays and moved the logic to cloudinary.url()